cdap-proto/src/main/java/io/cdap/cdap/proto/NamespaceRepositoryConfig.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.proto;
+
+import com.google.gson.annotations.JsonAdapter;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the repository configuration of a namespace. This class needs to be GSON serializable.
+ */
+@JsonAdapter(NamespaceRepositoryConfigCodec.class)
+public class NamespaceRepositoryConfig {
+  public static final String PROVIDER = "provider";
+  public static final String REPOSITORY_LINK = "repository.link";
+  public static final String DEFAULT_BRANCH = "default.branch";
+  public static final String AUTH_TYPE = "auth.type";
+  public static final String USERNAME = "username";
+  public static final String PATH_PREFIX = "path.prefix";
+
+  private final Map<String, String> configs;
+
+  public NamespaceRepositoryConfig(@Nullable String provider, @Nullable String repoLink, @Nullable String defaultBranch,
```

Comment:
same suggestions, some variables have 'repo' prefix some do not

Reply:
updated to make the vars more consistent